subworkflows/local/amr_annotation.nf

```diff
@@ -0,0 +1,103 @@
+// Importing the necessary modules for the workflow
+
+include { ABRITAMR_RUN } from '../../modules/nf-core/abritamr/run/main'
+include { RGI_MAIN } from '../../modules/nf-core/rgi/main/main'
+include { AMRFINDERPLUS_UPDATE } from '../../modules/nf-core/amrfinderplus/update/main'
+include { AMRFINDERPLUS_RUN } from '../../modules/nf-core/amrfinderplus/run/main'
+include { ABRICATE_RUN } from '../../modules/nf-core/abricate/run/main'
+include { ABRICATE_SUMMARY } from '../../modules/nf-core/abricate/summary/main'
+
+// This will be implemented in the future when the database
+// information is ready for hamronization tool to parse
+
+/*
+include { HAMRONIZATION_SUMMARIZE } from '../../modules/nf-core/hamronization/summarize/main'
+include { HAMRONIZATION_AMRFINDERPLUS } from '../../modules/nf-core/hamronization/amrfinderplus/main'
+include { HAMRONIZATION_RGI } from '../../modules/nf-core/hamronization/rgi/main'
+include { HAMRONIZATION_ABRICATE } from '../../modules/nf-core/hamronization/abricate/main'
+*/
+
+// Defining the workflow
+workflow AMR_ANNOTATION{
+    // Defining the input channel
+    take:
+        ch_genome
+
+    // Defining the main process
+    main:
+        // Initializing empty channels for the output files
+        ch_versions = Channel.empty()
+        abricate_report = Channel.empty()
+        abricate_summary_report = Channel.empty()
+        rgi_report = Channel.empty()
+        amrfinderplus_report = Channel.empty()
+        abritamr_report = Channel.empty()
+
+
+        // Running abricate
+        if (!params.skip_abricate){
+            abricate_dbs = Channel.empty()
+            if (params.abricate_db != null){
+                abricate_dbs = Channel.from(params.abricate_db.tokenize(","))
+            }
+            else{
+                abricate_dbs = []
+            }
+            ABRICATE_RUN(ch_genome, abricate_dbs)
+            abricate_report = ABRICATE_RUN.out.report
+            ch_versions = ABRICATE_RUN.out.versions
+            if (!params.skip_abricate_summary){
+                reports = abricate_report.collect()
+                ABRICATE_SUMMARY(reports)
+                abricate_summary_report = ABRICATE_SUMMARY.out.report
+                ch_versions = ch_versions.mix(ABRICATE_SUMMARY.out.versions)
+            }
+            /*
+            if (!params.skip_hamronization ){
+                HAMRONIZATION_ABRICATE(abricate_summary_report, "tsv", ABRICATE_RUN.out.versions, "test")
+            }
+            */
+        }
+
+        if (!params.skip_rgi){
+            RGI_MAIN(ch_genome)
+            rgi_report = RGI_MAIN.out.tsv
+            ch_versions = ch_versions.mix(RGI_MAIN.out.versions)
+            /*
+            if (!params.skip_hamronization ){
+                HAMRONIZATION_RGI(rgi_report, "tsv", RGI_MAIN.out.versions, "test")
+            }*/
+        }
+
+        if (!params.skip_amrfinderplus){
+            AMRFINDERPLUS_UPDATE()
```

Comment:
I don't think we want to pull the latest AMRFinderPlus database every time we run the AMR workflow. Perhaps only download the db when the user does not provide path to the db. Speaking of which, I think there should be a pipeline parameter to allow users to supply their own db.

Reply:
resolved with https://github.com/cidgoh/nf-seqqc/pull/47/commits/fac0242c06bc150a56a4f045521f7d78abd5a1f4